macros/src/utils.rs

```diff
@@ -29,11 +29,16 @@ macro_rules! impl_parse {
             fn parse($input: syn::parse::ParseStream) -> syn::Result<Self> {
                 let mut $out = $i::default();
                 loop {
+                    let span = $input.span();
```

Comment:
Getting the span before the key improves the location of the error:

## Assignment attributes
Before, points at the equals sign:
![image](https://github.com/Aleph-Alpha/ts-rs/assets/131818645/f57fefae-75aa-445b-abc7-0af7da02ee3c)

After, points at the attribute's name:
![image](https://github.com/Aleph-Alpha/ts-rs/assets/131818645/185adddd-a7b7-4f57-ae74-91da80ed1dfa)

## Flag attributes:
Before, points at the token after the attribute (close bracket or comma):
![image](https://github.com/Aleph-Alpha/ts-rs/assets/131818645/13a34f32-fa61-414c-8db9-a95f932b33de)

After, points at the attribute's name:
![image](https://github.com/Aleph-Alpha/ts-rs/assets/131818645/4e052b1c-78ad-4d1f-a178-60770dcfa1c3)

Reply:
Love it, great work!

Reply:
Thanks! Btw, this is the first good look I've had at exactly how `impl_parse!` works, and I gotta say, I'm definetly using that for every proc_macro I make in the future lol. Awesome job man!